src/qibo/models/variational.py

```diff
@@ -336,6 +336,20 @@ class QAOA(object):
 
     from qibo import hamiltonians, optimizers
 
+    def _cvar_loss(params, qaoa, hamiltonian, state):
+        if state is not None:
+            state = hamiltonian.backend.cast(state, copy=True)
+        qaoa.set_parameters(params)
+        state = qaoa(state)
+        return hamiltonian.expectation(state)
```

Comment:
thanks for spotting the mistake. :) I will use self as the first argument.

Reply:
As of now, my main concern is that cvar and gibbs would involve measurement of the quantum state. Hence it can be backend-dependent.  Any idea how can we bypass this challenge?